### PR TITLE
Automated cherry pick of #64296: Update nvidia-gpu-device-plugin to apps/v1 and use #64340: Fix DsFromManifest() after we switch from extensions/v1beta1 #64404: DaemonSet internals are still in extensions

### PR DESCRIPTION
**Pull request by jiayingz**
Cherry pick of #64296 #64340 #64404 on release-1.10.

#64296: Update nvidia-gpu-device-plugin to apps/v1 and use
#64340: Fix DsFromManifest() after we switch from extensions/v1beta1
#64404: DaemonSet internals are still in extensions

```release-note
Fix nvidia-gpu-device-plugin DaemonSet config to ensure correct device plugin upgrade
```

### CODE REVIEW

File: test/e2e/scheduling/BUILD [hunk 1]
```diff
@@ -22,6 +22,7 @@ go_library(
         "//pkg/api/v1/pod:go_default_library",
         "//pkg/apis/core:go_default_library",
         "//pkg/apis/extensions:go_default_library",
+        "//pkg/apis/scheduling:go_default_library",
```
Comment (mindprince) [inline test/e2e/scheduling/BUILD]: Why is this added?

**Review by rohitagarwal003 — APPROVED** [review]
/lgtm
